### DISCUSSION TITLE
[Features] More Aggregations

### DISCUSSION THREAD

**Discussion by jaychia** [Eventual-Inc/Daft#485] [Ideas]
# Summary

Aggregations are used in a few contexts:

1. Global aggregation (performed on the entire column of a DataFrame, or semantically equivalently - grouped by an empty set of keys `[]`)
2. Grouped aggregations (performed on columns grouped by certain keys)
3. Windows (rolling windows of computation)

Daft currently supports a few aggregations: `sum`, `count`, `mean`, `list`, `concat`, `min`, `max`.

Some of the aggregations we would like to support in the future include the following:

- [ ] std
- [ ] set - like `list`, but only keep distinct elements
- [ ] count_distinct - like `count`, but only count distinct elements
- [ ] quantiles

**Comment by ym-pett** [comment]
that would be great! could I add a few more descriptive statistics to this: 

- [ ] median 
- [ ] mode 
- [ ] rank
- [ ] kurtosis

For context, I'm collaborating on `narwhals-daft`, a plugin which makes `daft` DataFrames accessible to `Narwhals`. Having those extra descriptives would make it easier to implement and conform with what most of the other modules contain. Thanks!
